libs/platform/ebpf_object.c

```diff
@@ -128,26 +206,75 @@ ebpf_duplicate_utf8_string(_Out_ ebpf_utf8_string_t* destination, _In_ const ebp
     }
 }
 
+_Requires_lock_held_(&_ebpf_object_tracking_list_lock) static ebpf_object_t* _get_next_object(
```

Comment:
I don't think this always returns values in increasing order. Assume the following:
Valid IDs are:
0x1, 0x3, 0x10002

They will have indexes:
1, 2, and 3 respectively. 

_get_next_object called with start_id = 0x0.

This will return:
0x1, 0x10002, and 3.

Reply:
Misunderstanding on my part. The index is the high bit, the counter is the lower bit.